dotcom-rendering/src/web/components/Portal.tsx

```diff
@@ -1,26 +1,31 @@
-import ReactDOM from 'react-dom';
+import { Placeholder } from './Placeholder';
 
-type Props = {
-	rootId: string;
-	children: React.ReactNode;
-};
+interface Props {
+	componentName: string;
+	when?: 'immediate' | 'idle' | 'visible';
+	props?: any;
```

Comment:
Is there a way I can use `componentName` to get the expected type and use that here? This seems pretty wild but would really help improve the dev ex!

Reply:
@JamieB-gu @mxdvl ?

Reply:
You mean read it from the existing files? Yes, but you’d have to have some sort of process that generates a declaration file from reading the filesystem, I think.

That’s what the interview tool does in [`build.ts`](https://github.com/guardian/interview-questions/blob/main/build.ts)

Reply:
I don’t think it’s a lot of work, and that file has another added benefits: It’s a live document containing all the current portals. I’m happy to set half-an-hour aside to have a crack at this.

The question that remains is how the generation steps gets called, but I’m sure there are precendents… `make portals`?

Reply:
🤔 I'm not 100% following your intention but would what you're thinking of solve the dev ex problem? That is, I want to insert a new Portal and type

<Portal componentName="MyThing" ...

And then when I go to enter the `props` they are already typed as needed?